Funcky/Extensions/ParseExtensions/ParseExtensions.AssemblyNameInfo.cs

```diff
@@ -0,0 +1,9 @@
+#if REFLECTION_ASSEMBLY_NAME_INFO
+using System.Reflection.Metadata;
+using Funcky.Internal;
+
+namespace Funcky.Extensions;
+
+[OrNoneFromTryPattern(typeof(AssemblyNameInfo), nameof(AssemblyNameInfo.TryParse))]
```

Comment:
I feel `AssemblyNameInfo` is fairly niche, should we really add the parse methods?

Reply:
I think we just do it, why adding some additional criteria. If it is in there, we support it.